lib/settings/wifi_screen.dart

```diff
@@ -48,67 +57,122 @@ class _WifiScreenState extends State<WifiScreen> {
-        -90: Icon(Icons.warning_rounded, color: Colors.red[300]),
-      };
+    final Map<int, Icon> otherIcons = {
+      10: Icon(Icons.network_wifi_rounded, color: Colors.green[300], size: 30),
+      0: Icon(Icons.network_wifi_rounded, color: Colors.green[300], size: 30),
+      -50: Icon(Icons.network_wifi_3_bar_rounded,
+          color: Colors.green[300], size: 30),
+      -70: Icon(Icons.network_wifi_2_bar_rounded,
+          color: Colors.orange[300], size: 30),
+      -90: Icon(Icons.warning_rounded, color: Colors.red[300], size: 30),
+    };
 
-      signalStrength = signalStrength.abs();
-      for (var threshold in icons.keys.toList().reversed) {
-        if (signalStrength >= threshold.abs()) {
-          return icons[threshold]!;
-        }
+    // Choose the correct map of icons based on the platform
+    final Map<int, Icon> icons = platform == 'linux' ? linuxIcons : otherIcons;
+
+    // Find the first icon where the signal strength is less than or equal to the threshold
+    for (var threshold in icons.keys.toList().reversed) {
+      if (signalStrength <= threshold) {
+        return icons[threshold]!;
       }
     }
+
+    // If no icon was found, return a warning icon
     return const Icon(Icons.warning_rounded);
   }
 
   late String platform;
-  Future<List<Map<String, String>>> _getWifiNetworks() async {
+
+  Future<List<Map<String, String>>> _getWifiNetworks(
+      {bool alreadyConnected = false}) async {
     wifiNetworks.clear();
     try {
+      /*if (Theme.of(context).platform == TargetPlatform.macOS &&
+          !alreadyConnected) {
+        currentWifiSSID = 'test';
+      }*/
       ProcessResult? result;
-      ProcessResult? currentSSID;
       switch (Theme.of(context).platform) {
         case TargetPlatform.macOS:
           platform = 'macos';
-          final List<String> networks =
-              await macplatform.invokeMethod('networks');
-          currentSSID =
-              await Process.run('networksetup', ['-getairportnetwork', 'en0']);
-          final match = RegExp(r'Current Wi-Fi Network: (.*)')
-              .firstMatch(currentSSID.stdout.toString());
-          currentWifiSSID = match?.group(1)?.trim() ?? '';
-          return networks.map((ssid) => {'SSID': ssid, 'SIGNAL': '0'}).toList();
+          final List<Map<String, String>> networks = [];
+          // Generate fake output
+          for (int i = 0; i < 10; i++) {
+            int rand = Random().nextInt(100);
+            networks.add({
+              'SSID': 'Network $rand',
+              'SIGNAL':
+                  '${-30 - i * 5}', // Signal strength decreases with each network
+              'BSSID': '00:0a:95:9d:68:1$i',
+              'RSSI': '${-30 - i * 5}',
+              'CHANNEL': '${1 + i}',
+              'HT': 'Y',
+              'CC': 'US',
+              'SECURITY': '(WPA2)'
+            });
+          }
+          //if (!alreadyConnected) currentWifiSSID = networks.first['SSID'];
```

Comment:
and this?

Reply:
Resolved by [17dd19d](https://github.com/TheContrappostoShop/Orion/pull/14/commits/17dd19d100d306003192341c4a4ca3d23aad545d)